### DISCUSSION TITLE
to_timestamp should return a timestamp with time zone

### DISCUSSION THREAD

**Discussion by xcaptain** [duckdb/duckdb#6391] [Ideas]
Currently `epoch_ms` and `to_timestamp` have the same effect, is a little duplicated.

CLI:

```sql
select to_timestamp(1676764800) as tz1, epoch_ms(1676764800000) as tz2;
```

got:
```
┌─────────────────────┬─────────────────────┐
│         tz1         │         tz2         │
│      timestamp      │      timestamp      │
├─────────────────────┼─────────────────────┤
│ 2023-02-19 00:00:00 │ 2023-02-19 00:00:00 │
└─────────────────────┴─────────────────────┘
```

I suggest to let `to_timestamp` return a timestamp with timezone like postgres does, would be more helpful.

Postgres:

```sql
select to_timestamp(1676764800) as tz, to_timestamp(1676764800) at time zone 'UTC' as tz1, to_timestamp(1676764800) at time zone 'Asia/Shanghai' as tz2;
```

got:
```
tz                           |tz1                    |tz2                    |
-----------------------------+-----------------------+-----------------------+
2023-02-19 08:00:00.000 +0800|2023-02-19 00:00:00.000|2023-02-19 08:00:00.000|
```

 `to_timestamp` now return a timestamp without timezone, when query with `at time zone xxx`, the convert result is wrong, if return a timestamptz, I think the problem will disappear

**Comment by xcaptain** [comment]
It's easy to change, just modify one line https://github.com/duckdb/duckdb/blob/master/src/function/scalar/date/epoch.cpp#L42
change `LogicalType::TIMESTAMP` to `LogicalType::TIMESTAMP_TZ`, but some tests must also change.

**Comment by xcaptain** [comment]
@Mytherin What do you think

**Comment by Mytherin** [comment]
The problem with pushing anything that returns a `TIMESTAMPTZ` into the main library is that `TIMESTAMPTZ` cannot be operated on without `ICU` installed. The function would have to be moved to the `ICU` library. Perhaps we could add a function `to_timestamptz` to the ICU extension instead?

**Comment by xcaptain** [comment]
How about make icu a builtin feature rather than an extension? I usually use `TimestampTZ` because it's much convenient when doing query, do you recommend to prefer `Timestamp`?

As for the naming of `to_timestamptz`, I prefer to use `to_timestamp` because postgres does.
https://www.postgresql.org/docs/current/functions-formatting.html
![image](https://user-images.githubusercontent.com/4054836/220506939-1633dfb9-4ba5-4ff9-8022-7e266b15b8c1.png)


Also I found another difference between duckdb and postgres in timestamp;

CLI:
```sql
select '2023-02-19 08:00:00+08'::TIMESTAMP;
```

got:
```
┌─────────────────────────────────────────────┐
│ CAST('2023-02-19 08:00:00+08' AS TIMESTAMP) │
│                  timestamp                  │
├─────────────────────────────────────────────┤
│ 2023-02-19 00:00:00                         │
└─────────────────────────────────────────────┘
```

While in Postgres:

```sql
select '2023-02-19 08:00:00+08'::TIMESTAMP;
```

got:

```
timestamp              |
-----------------------+
2023-02-19 08:00:00.000|
```

The difference is because postgres will ignore `+`/`-` part in a string literal, see: https://www.postgresql.org/docs/current/datatype-datetime.html
![image](https://user-images.githubusercontent.com/4054836/220505705-ec01ecf1-f585-49af-b1e7-b56361aeda8d.png)

**Reply by Mytherin** [reply]
Icu is an extension because it more than doubles the binary size for a set of features that not everybody needs. That is why we want to keep it optional.

**Reply by xcaptain** [reply]
Ok, I will change my field types to `TIMESTAMP` to avoid these small problems.
